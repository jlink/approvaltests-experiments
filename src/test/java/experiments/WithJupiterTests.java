package experiments;

import java.util.*;

import org.approvaltests.*;
import org.approvaltests.combinations.*;
import org.junit.jupiter.api.*;

class WithJupiterTests {

	@Test
	void testString() {
		Approvals.verify("hello approvals");
	}

	@Test
	void testList() {
		String[] names = {"Llewellyn", "James", "Dan", "Jason", "Katrina"};
		Arrays.sort(names);
		Approvals.verifyAll("", names);
	}

	@Test
	void combinatorics() {
		CombinationApprovals.verifyAllCombinations(
			(a, b) -> a + "=" + b,
			new String[]{"a", "b", "c"},
			new String[]{"1", "2", "3"}
		);
	}
}
