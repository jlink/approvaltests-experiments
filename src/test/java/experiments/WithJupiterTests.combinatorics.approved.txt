[a, 1] => a=1 
[a, 2] => a=2 
[a, 3] => a=3 
[b, 1] => b=1 
[b, 2] => b=2 
[b, 3] => b=3 
[c, 1] => c=1 
[c, 2] => c=2 
[c, 3] => c=3 
